cdap-ui/app/directives/log-viewer/log-viewer.js

```diff
@@ -270,6 +273,55 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi, LOGVIEWERSTORE_
     });
   };
 
+  var exportTimeout = null;
+
+  const downloadLogs = () => {
+    return myLogsApi.getLogsStartAsRaw({
+          'namespace' : this.namespaceId,
+          'appId' : this.appId,
+          'programType' : this.programType,
+          'programId' : this.programId,
+          'runId' : this.runId,
+          'start' : this.startTimeSec
+    }).$promise.then(
+      (res) => {
+        this.downloadContent = res;
+      },
+      (err) => {
+        console.log('ERROR: ', err);
+        return;
```

Comment:
Don't think we need a return statement. We should surface errors if any while fetching the logs for download. User should get a feedback if something goes wrong.